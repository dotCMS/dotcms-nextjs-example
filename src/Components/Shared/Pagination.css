.pagination {
    border-radius: 0.25rem;
    display: inline-block;
    margin: 1.25rem 0;
    padding-left: 0;
}

.pagination .page-item {
    display: inline-block;
}

.pagination li button {
    background: var(--white);
    border: 1px solid #e6e6e6;
    border-radius: 0.125rem;
    color: #000;
    line-height: 1.42857;
    margin: 0.125rem;
    padding: 0.625rem 1.25rem;
    text-decoration: none;
}

.pagination li.active button.page-link {
    background: var(--background-color);
    border: 1px solid var(--background-color);
}

.pagination li button:hover,
.pagination li button:focus {
    background: var(--background-color);
    border: 1px solid var(--background-color);
    color: var(--white);
    transition: 0.5s;
}
