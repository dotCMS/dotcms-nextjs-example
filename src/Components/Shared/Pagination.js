import React from 'react';
import PropTypes from 'prop-types';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';

export default class DotPagination extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      activePage: 0,
      totalItems: 0
    };
  }

  componentDidMount() {
    this.setState(state => ({
      ...state,
      totalItems: this.props.totalItems
    }));
  }

  pageChange(pageNumber) {
    this.setState(state => ({
      ...state,
      activePage: pageNumber
    }));

    this.props.onPageChange(pageNumber);
  }

  render() {
    const { activePage, totalItems } = this.state;
    const { pageSize } = this.props;
    const pages = Math.ceil(totalItems / pageSize) || 0;
    return (
      <>
        <Pagination className="dot-pagination" aria-label="Page navigation">
          {[...Array(pages)].map((x, i) => (
            <PaginationItem key={i} active={activePage === i}>
              <PaginationLink onClick={() => this.pageChange(i)}>
                {i + 1}
              </PaginationLink>
            </PaginationItem>
          ))}
        </Pagination>
      </>
    );
  }
};

DotPagination.propTypes = {
  pageSize: PropTypes.number,
  totalItems: PropTypes.number
};
