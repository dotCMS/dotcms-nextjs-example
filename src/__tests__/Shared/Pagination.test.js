import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../../Components/Shared/Pagination';
import toJSON from 'enzyme-to-json';

describe('<Pagination />', () => {
    it('renders pagination correctly', () => {
        const pageSize = 9;
        const totalItems = 25;
        const wrapper = shallow(
            <Pagination
                pageSize={pageSize}
                totalItems={totalItems}
                onPageChange={() => {}}
            />
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should call and send page number', async () => {
        const pageSize = 9;
        const totalItems = 25;
        const pageChange = jest.fn();
        const wrapper = shallow(
            <Pagination
                pageSize={pageSize}
                totalItems={totalItems}
                onPageChange={pageChange}
            />
        );
        wrapper
            .find('PaginationLink')
            .at(1)
            .simulate('click');
        expect(pageChange).toHaveBeenCalledWith(1);
        expect(wrapper.state().activePage).toBe(1);
    });
});
