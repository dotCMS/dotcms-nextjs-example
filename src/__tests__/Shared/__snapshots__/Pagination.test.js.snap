// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders pagination correctly 1`] = `
<Fragment>
  <Pagination
    aria-label="Page navigation"
    className="dot-pagination"
    listTag="ul"
    tag="nav"
  >
    <PaginationItem
      active={true}
      key="0"
      tag="li"
    >
      <PaginationLink
        onClick={[Function]}
        tag="a"
      >
        1
      </PaginationLink>
    </PaginationItem>
    <PaginationItem
      active={false}
      key="1"
      tag="li"
    >
      <PaginationLink
        onClick={[Function]}
        tag="a"
      >
        2
      </PaginationLink>
    </PaginationItem>
    <PaginationItem
      active={false}
      key="2"
      tag="li"
    >
      <PaginationLink
        onClick={[Function]}
        tag="a"
      >
        3
      </PaginationLink>
    </PaginationItem>
  </Pagination>
</Fragment>
`;
